web/src/lib/util.ts

```
@@ -23,6 +23,83 @@ export function trimCommitHash(hash: string): string {
     return hash;
 }
 
+export async function isBinaryFile(file: File): Promise<boolean> {
```

if there's a mime, you should fail early